core/src/main/scala/org/apache/spark/internal/io/SparkHadoopMapReduceWriter.scala

```diff
@@ -163,12 +164,20 @@ object SparkHadoopMapReduceWriter extends Logging {
             outputMetricsAndBytesWrittenCallback, recordsWritten)
           recordsWritten += 1
         }
-
+        writer.close(taskContext)
```

Comment:
Ah, but do you still want to null it here so that it doesn't also cause a close() in the catch block? the commit could fail.